service/matching/tasklist/task_list_manager.go

```diff
@@ -274,6 +275,14 @@ func (c *taskListManagerImpl) Start() error {
 	if c.taskListID.IsRoot() && c.taskListKind == types.TaskListKindNormal {
 		c.partitionConfig = c.db.PartitionConfig().ToInternalType()
 		c.logger.Info("get task list partition config from db", tag.Dynamic("root-partition", c.taskListID.GetRoot()), tag.Dynamic("task-list-partition-config", c.partitionConfig))
+		if c.partitionConfig != nil {
+			// push update notification to all non-root partitions on start
+			c.stopWG.Add(1)
+			go func() {
+				defer c.stopWG.Done()
+				c.notifyPartitionConfig(context.Background(), *c.partitionConfig, int(c.partitionConfig.NumReadPartitions))
```

Comment:
nit: we should add a `ctx` and `cancelFn` pair to tasklist manager to better respond to shutdown events and avoid using background context for this kind of background stuff